x/dyncomm/ante/ante.go

```diff
@@ -0,0 +1,84 @@
+package ante
+
+import (
+	"fmt"
+
+	dyncommkeeper "github.com/classic-terra/core/v2/x/dyncomm/keeper"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// DyncommDecorator checks for EditValidator and rejects
+// edits that do not conform with dyncomm
+type DyncommDecorator struct {
+	dyncommKeeper dyncommkeeper.Keeper
+	stakingKeeper stakingkeeper.Keeper
+}
+
+func NewDyncommDecorator(dk dyncommkeeper.Keeper, sk stakingkeeper.Keeper) DyncommDecorator {
+	return DyncommDecorator{
+		dyncommKeeper: dk,
+		stakingKeeper: sk,
+	}
+}
+
+// IsMsgSubmitProposal checks whether the input msg is a MsgSubmitProposal
+func IsMsgEditValidator(msg sdk.Msg) bool {
+	_, ok := msg.(*stakingtypes.MsgEditValidator)
+	return ok
+}
+
+func (dd DyncommDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	if simulate {
+		return next(ctx, tx, simulate)
+	}
+
+	msgs := tx.GetMsgs()
+	err := dd.FilterMsgsAndCheckEditValidator(ctx, msgs...)
```

Comment:
This can be resolved. At the time of `CreateValidator` it's not completely know how much voting power it's going to have. It's very likely that upon creation most Validators are in the inactive set anyway. When they enter the active set their commissions will be updated anyway as soon as the epoch hits end.